plugins/fortinet_fortigate/icon_fortinet_fortigate/util/api.py

```diff
@@ -0,0 +1,104 @@
+        self.session = Session()
+        self.session.params = {"access_token": api_key}
+        self.helper = Helpers(self.logger)
+
+    def call_api(
+        self, path: str, method: str = "GET", params: dict = None, json_data: dict = None
+    ) -> requests.Response:
+        try:
+            response = self.session.request(
+                method=method.upper(),
+                url=f"https://{self.host}/api/v2/cmdb/{path}",
+                verify=self.ssl_verify,
+                json=json_data,
+                params=params,
+            )
+            self.helper.http_errors(response.text, response.status_code)
+            if response.status_code > 500:
+                raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response.text)
+
+            if 200 <= response.status_code < 300:
+                return response.json()
+
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
+        except json.decoder.JSONDecodeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=e)
+        except requests.exceptions.HTTPError as e:
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)
+
+    def create_address_object(self, name: str, host: str, address_type: str):
+        endpoint = "firewall/address"
+        payload = {"name": name, "type": address_type}
+        if address_type == "ipmask":
+            payload["subnet"] = host
+        elif address_type == "fqdn":
+            payload["fqdn"] = host
+        else:
+            payload["ip6"] = host
+            endpoint = "firewall/address6"
+        return self.call_api(endpoint, "POST", json_data=payload)
+
+    def delete_address_object(self, endpoint: str):
+        return self.call_api(endpoint, "DELETE")
+
+    def get_address_group(self, address_group_name, is_ipv6):
+        endpoint = "firewall/addrgrp"
+
+        if is_ipv6:
+            endpoint = "firewall/addrgrp6"
+
+        result = self.call_api(
+            path=endpoint,
+            params={
+                "filter": f"name=@{address_group_name}",
+            },
+        )
+
+        groups = result.get("results")
+        if not len(groups) > 0:
+            raise PluginException(
+                cause=f"Could not find address group '{address_group_name}' in results.\n",
+                assistance=f"Please make sure the group '{address_group_name}' exists.\n",
+                data=result,
+            )
+
+        return groups[0]
+
+    def get_address_object(self, address_name):
+        try:
+            response_ipv4 = self.call_api(path=f"firewall/address/{address_name}")
+            if response_ipv4.get("http_status") == 200:
+                return response_ipv4
+        except (PluginException, json.decoder.JSONDecodeError, requests.exceptions.HTTPError):
+            pass
+
+        response_ipv6 = self.call_api(path=f"firewall/address6/{address_name}")
```

Comment:
@pmara-r7 Based on the name of the address object, we are not able to tell whether it is an IPv4 or IPv6 object, so this method is used to determine whether a given object belongs to an IPv4 or IPv6 objects and what endpoint should be used when working with this object. 

Another solution instead of trying to get this object directly is to get all IPv4 address objects and look for a specific object by name in them. If it is not found then we would have to get all the IPv6 address objects and search again to find the specific object. This would avoid such use of exceptions, but extend the execution time of the actions where this method is used.

Reply:
Perhaps we can come to a middle ground here- which of the exceptions is thrown when we have this kind of "search failure". That way we can only ignore the one that is expected to be produced. Or are you saying it's possible for all 3 of these to be raised? I'd just really appreciate that we only ignore exceptions we are actually expecting to happen.

Reply:
(this is the last "unresolved comment" I have by the way)

Reply:
Ok, then we can only ignore PluginException here and leave the handling of the rest of the exceptions unchanged.